runtime/calamari/src/diff_gas_fees.rs

```diff
@@ -0,0 +1,2169 @@
+            pallet_parachain_staking::Call::cancel_leave_candidates { candidate_count: 2 },
+        );
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "cancel_leave_candidates",
+            dispatch_info,
+            call_len,
+        ));
+
+        // go_offline
+        let call = crate::Call::ParachainStaking(pallet_parachain_staking::Call::go_offline {});
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "go_offline",
+            dispatch_info,
+            call_len,
+        ));
+
+        // go_offline
+        let call = crate::Call::ParachainStaking(pallet_parachain_staking::Call::go_offline {});
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "go_offline",
+            dispatch_info,
+            call_len,
+        ));
+
+        // candidate_bond_more
+        let call =
+            crate::Call::ParachainStaking(pallet_parachain_staking::Call::candidate_bond_more {
+                more: 1,
+            });
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "candidate_bond_more",
+            dispatch_info,
+            call_len,
+        ));
+
+        // schedule_candidate_bond_less
+        let call = crate::Call::ParachainStaking(
+            pallet_parachain_staking::Call::schedule_candidate_bond_less { less: 1 },
+        );
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "schedule_candidate_bond_less",
+            dispatch_info,
+            call_len,
+        ));
+
+        // execute_candidate_bond_less
+        let call = crate::Call::ParachainStaking(
+            pallet_parachain_staking::Call::execute_candidate_bond_less { candidate: ALICE },
+        );
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "execute_candidate_bond_less",
+            dispatch_info,
+            call_len,
+        ));
+
+        // cancel_candidate_bond_less
+        let call = crate::Call::ParachainStaking(
+            pallet_parachain_staking::Call::cancel_candidate_bond_less {},
+        );
+        let (dispatch_info, call_len) = get_call_details(&call);
+        calamari_runtime_calls.push((
+            "pallet_parachain_staking",
+            "cancel_candidate_bond_less",
+            dispatch_info,
+            call_len,
+        ));
+
+        // delegate
+        let call = crate::Call::ParachainStaking(pallet_parachain_staking::Call::delegate {
```

Comment:
This method is great, but it can't substitute weight comparison as it is.

In this TX you check for almost the cheapes TX possible and guarantee that this *specific* tx signature is not changed.
What about a delegate call with `candidate_delegation_count: 5000` and `delegation_count:10000` ?
This is part of the dynamic component of the weights and changes in that are not captured this way, not allowing to determine whether the extrinsic with above parameters has changed in weight or not.

To get "good enough"-level coverage i guess we could have multiple invocations per extrinsic
- the "base case" simplest possible call as it is now
- One call with each of the dynamic weight components ( as specified in the benchmarking fn for this extrinsic) at its max/a very large setting
- Technically we'd also need all permutations of high/low values for each dynamic parameter, but that explodes the  complexity so i'd recommend to leave that out

For this example this would be 
- pallet_parachain_staking::Call::delegate(ALICE, 1,2,3)
- pallet_parachain_staking::Call::delegate(ALICE, 1,25,3)
- pallet_parachain_staking::Call::delegate(ALICE, 1,2,100)
- (optionally, but leave it out) pallet_parachain_staking::Call::delegate(ALICE, 1,25,100)

Sry for this additional effort :|